src/request.ts

```diff
@@ -23,7 +23,7 @@ async function _request({
 	requestBody,
 	headers: requestHeaders,
 	...options
-}: RequestOptions): Promise<Response> {
+}: RequestOptions): Promise<unknown> {
```

Comment:
Is unknown the best option for a type here? Would love your thoughts on that

Reply:
Here are my thoughts on this:

1. This is a private function (not exported and denoted with an _ to further emphasise this)
2. It is used by the exported utility function `request` which has an unbounded `T`
3. `response.json()` returns `Promise<any>`
4. `any` is typically discouraged, many linters warn about it. `unknown` is the "type safe" alternative, it makes no assumptions about the type but doesn't give you the freedom to make your own assumptions about it either -- which would lead to a runtime error.
5. An alternative type could be `object`, but the response could technically be a number which is valid JSON I think.
6. The method that is given the type `T` will be able to cast with no resistance.

Did you have another type in mind?

Reply:
Casting as unknown also has no runtime cost, it's all thrown away after type checking and compilation.

Reply:
i was thinking maybe something like `Either response or error` but maybe that makes things too complicated.
```
type Only<T, U> = {
    [P in keyof T]: T[P];
} & {
    [P in keyof U]?: never;
};

type Either<T, U> = Only<T, U> | Only<U, T>
```

Not that important, was just curious to pick your brain